### PR TITLE
Remove all usages of dbconn.Global in registry

### PR DESCRIPTION
**Pull request by eseliger**
Refactored dbutil.DB to database.DB and got rid of all dbconn.Global references in the registry code.

25 remaining, mostly in the setup logic for dbconn.Global :)

### CODE REVIEW

File: enterprise/cmd/frontend/internal/registry/extension_bundle.go [hunk 1]
```diff
@@ -29,86 +29,88 @@ var sourceMappingURLLineRegex = lazyregexp.New(`(?m)\r?\n?^//# sourceMappingURL=
 
 // handleRegistryExtensionBundle serves the bundled JavaScript source file or the source map for an
 // extension in the registry as a raw JavaScript or JSON file.
-func handleRegistryExtensionBundle(w http.ResponseWriter, r *http.Request) {
-	if conf.Extensions() == nil {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
+func handleRegistryExtensionBundle(db database.DB) func(w http.ResponseWriter, r *http.Request) {
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/registry/extension_bundle.go:32]: same nit:
```suggestion
func handleRegistryExtensionBundle(db database.DB) http.HandlerFunc {
 ```

File: enterprise/cmd/frontend/internal/registry/publisher_graphql.go [hunk 2]
```diff
@@ -136,14 +135,14 @@ func unmarshalRegistryPublisherID(id graphql.ID) (*registryPublisherID, error) {
...
+func (p *registryPublisherID) viewerCanAdminister(ctx context.Context, db database.DB) error {
 	switch {
 	case p.userID != 0:
 		// 🚨 SECURITY: Check that the current user is either the publisher or a site admin.
-		return backend.CheckSiteAdminOrSameUser(ctx, database.NewDB(db), p.userID)
+		return backend.CheckSiteAdminOrSameUser(ctx, db, p.userID)
 	case p.orgID != 0:
 		// 🚨 SECURITY: Check that the current user is a member of the publisher org.
-		return backend.CheckOrgAccessOrSiteAdmin(ctx, database.NewDB(db), p.orgID)
+		return backend.CheckOrgAccessOrSiteAdmin(ctx, db, p.orgID)
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/registry/publisher_graphql.go:145]: Nice! Glad to see more of these `databsae.NewDB()` going away. I'd love to get to the point where we can just call that once with a `*sql.DB` in our main func and get rid of `dbutil.DB` entirely.

**Review by camdencheek — APPROVED** [review]
Woo hoo! Thanks for this, this is big progress towards getting rid of `dbconn.Global` entirely.

File: enterprise/cmd/frontend/internal/registry/publisher_graphql.go (hunk 2 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/registry/publisher_graphql.go:145]: Same!

**Review by mrnugget — APPROVED** [review]
Well done!

**Pull request merged**